lib/msf/base/sessions/powershell.rb

```diff
@@ -0,0 +1,319 @@
+# -*- coding: binary -*-
+require 'msf/base'
+require 'msf/base/sessions/scriptable'
+require 'shellwords'
+
+module Msf
+module Sessions
+
+###
+#
+# This class provides basic interaction with a command shell on the remote
+# endpoint.  This session is initialized with a stream that will be used
+# as the pipe for reading and writing the command shell.
+#
+###
+class PowerShell
+
+  #
+  # This interface supports basic interaction.
+  #
+  include Msf::Session::Basic
+
+  #
+  # This interface supports interacting with a single command shell.
+  #
+  include Msf::Session::Provider::SingleCommandShell
+
+  include Msf::Session::Scriptable
+
+
+  ##
+  # :category: Msf::Session::Scriptable implementors
+  #
+  # Executes the supplied script, must be specified as full path.
+  #
+  # Msf::Session::Scriptable implementor
+  #
+  def execute_file(full_path, args)
+    o = Rex::Script::Shell.new(self, full_path)
+    o.run(args)
+  end
+
+  #
+  # Returns the type of session.
+  #
+  def self.type
+    "powershell"
```

Comment:
Given that the only change here is the the session type, instead of duplicating this session class, it should inherit from the Shell session and then override the self.type method.